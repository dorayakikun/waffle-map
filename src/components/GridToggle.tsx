import * as React from 'react'
import { Checkbox, CheckboxProps } from 'semantic-ui-react'

export interface Props {
  title: string
  isShowGrid: boolean
  onToggleChanged: (
    event: React.FormEvent<HTMLInputElement>,
    data: CheckboxProps
  ) => void
}

export const GridToggle = (props: Props) => (
  <div>
    <div style={{ marginTop: '10px', marginBottom: '10px' }}>
      <span style={{ color: 'white', marginRight: 5 }}>{props.title}</span>
      <Checkbox
        checked={props.isShowGrid}
        onChange={props.onToggleChanged}
        toggle
      />
    </div>
  </div>
)
