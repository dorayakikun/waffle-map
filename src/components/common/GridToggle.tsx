import { FormControl, FormLabel, Switch } from "@chakra-ui/react";
import * as React from "react";

export type Props = {
  enableGrid: boolean;
  id: string;
  handleChanged: (event: React.FormEvent<HTMLInputElement>) => void;
  title: string;
};

export const GridToggle = (props: Props) => {
  return (
    <FormControl id={props.id} display={"flex"} alignItems={"center"}>
      <FormLabel htmlFor={props.title} mb={"0"}>
        {props.title}
      </FormLabel>
      <Switch
        id={props.title}
        isChecked={props.enableGrid}
        onChange={props.handleChanged}
      />
    </FormControl>
  );
};
