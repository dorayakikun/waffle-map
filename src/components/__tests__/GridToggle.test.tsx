import * as React from "react";
import * as render from "react-test-renderer";
import { GridToggle, Props } from "../GridToggle";

test("Should set props to MeshToggle", () => {
  const isShowGrid = false;
  const props: Props = {
    isShowGrid,
    onToggleChanged: () => {},
    title: "Show grid"
  };
  const tree = render.create(<GridToggle {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Should call onToggleChanged when changed checked", () => {
  const isShowGrid = false;
  const onToggleChanged = () => {};
  const props: Props = {
    isShowGrid,
    onToggleChanged,
    title: "Show grid"
  };
  const tree = render.create(<GridToggle {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
