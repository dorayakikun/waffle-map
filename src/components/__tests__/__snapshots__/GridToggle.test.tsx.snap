// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should call onToggleChanged when changed checked 1`] = `
<div>
  <div
    style={
      Object {
        "marginBottom": "10px",
        "marginTop": "10px",
      }
    }
  >
    <span
      style={
        Object {
          "color": "white",
          "marginRight": 5,
        }
      }
    >
      Show grid
    </span>
    <div
      className="ui fitted toggle checkbox"
      onChange={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <input
        checked={false}
        className="hidden"
        readOnly={true}
        tabIndex={0}
        type="checkbox"
      />
      <label />
    </div>
  </div>
</div>
`;

exports[`Should set props to MeshToggle 1`] = `
<div>
  <div
    style={
      Object {
        "marginBottom": "10px",
        "marginTop": "10px",
      }
    }
  >
    <span
      style={
        Object {
          "color": "white",
          "marginRight": 5,
        }
      }
    >
      Show grid
    </span>
    <div
      className="ui fitted toggle checkbox"
      onChange={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <input
        checked={false}
        className="hidden"
        readOnly={true}
        tabIndex={0}
        type="checkbox"
      />
      <label />
    </div>
  </div>
</div>
`;
